### PR TITLE
fix(src): ensure that querybuilder params are correctly used

### LINKED ISSUE #95: src parameter and query builder don't mix
```bash
# works
$ http "https://adobeioruntime.net/api/v1/web/helix-pages/helix-services/data-embed@v1/https://adobe.sharepoint.com/sites/TheBlog/_layouts/15/Doc.aspx?sourcedoc=%7BE7BF6B3A-F323-4076-968B-30A1565F2373%7D&file=query-index.xlsx&action=default&mobileredirect=true&hlx_p.limit=1" --print Hhb

# doesn't (returns all records)
http "https://adobeioruntime.net/api/v1/web/helix-pages/helix-services/data-embed@v1?src=https%3A%2F%2Fadobe.sharepoint.com%2Fsites%2FTheBlog%2F_layouts%2F15%2FDoc.aspx%3Fsourcedoc%3D%257BE7BF6B3A-F323-4076-968B-30A1565F2373%257D%26file%3Dquery-index.xlsx%26action%3Ddefault%26mobileredirect%3Dtrue&hlx_p.limit=1" --print Hhb
```

https://github.com/adobe/helix-content-proxy/issues/55

### PR DESCRIPTION
**Pull request by tripodsan**
fixes #95

### CODE REVIEW

File: src/data-source.js [hunk 1]
```diff
@@ -20,36 +20,46 @@ const querystring = require('querystring');
...
+    url = new URL(path.substring(1));
   }
-  const url = new URL(path.substring(1));
 
   if (!params.__ow_query) {
     // reconstruct __ow_query
+    const q = {};
     Object.keys(params)
       .filter((key) => !/^[A-Z]+_[A-Z]+/.test(key))
       .filter((key) => key !== 'api')
```
Comment (trieloff) [inline src/data-source.js:44]:
```suggestion
```

Comment (trieloff) [reply src/data-source.js:44]: Not needed anymore.

**Comment by tripodsan** [comment]
> Please add a post-deploy test. I don't trust the unit tests alone.

ok. I will add them...also it seems that the tests take a very long time to execute... do you know why?

**Comment by trieloff** [comment]
The tests are fast for me.

**Comment by tripodsan** [comment]
@trieloff I added a post-deploy test with the problematic format.

**Pull request merged**